packages/core/src/request.ts

```diff
@@ -313,6 +321,18 @@ export class Request<UserData extends Dictionary = Dictionary> {
         }
     }
 
+    private get enqueueStrategy(): EnqueueLinksOptions['strategy'] | undefined {
+        return this.userData.__crawlee?.enqueueStrategy;
+    }
+
+    private set enqueueStrategy(value: EnqueueLinksOptions['strategy']) {
+        if (!this.userData.__crawlee) {
+            (this.userData as Dictionary).__crawlee = { enqueueStrategy: value };
+        } else {
+            this.userData.__crawlee.enqueueStrategy = value;
+        }
+    }
```

Comment:
i guess we could have a helper for this part, as we do the same thing 4 times already now